src/main/java/org/junit/Assert.java

```diff
@@ -955,4 +955,48 @@ public static void assertEquals(Object[] expecteds, Object[] actuals) {
             Matcher<? super T> matcher) {
         MatcherAssert.assertThat(reason, actual, matcher);
     }
+
+    /**
+     * This interface facilitates the use of expectThrows from Java 8. It allows method references
+     * to void methods (that declare checked exceptions) to be passed directly into expectThrows
+     * without wrapping. It is not meant to be implemented directly.
+     *
+     * @since 4.13
+     */
+    public interface ThrowingRunnable {
+        void run() throws Throwable;
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception of type {@code throwableClass} when
+     * executed. If it does, the exception object is returned. If it does not throw an exception, an
+     * {@link AssertionError} is thrown. If it throws the wrong type of exception, an {@code
+     * AssertionError} is thrown describing the mismatch; the exception that was actually thrown can
+     * be obtained by calling {@link AssertionError#getCause}.
+     *
+     * @param throwableClass the expected type of the exception
+     * @param runnable       A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     * @since 4.13
+     */
+    public static <T extends Throwable> T expectThrows(Class<T> throwableClass, ThrowingRunnable runnable) {
+        try {
+            runnable.run();
+        } catch (Throwable t) {
+            if (throwableClass.isInstance(t)) {
+                return throwableClass.cast(t);
```

Comment:
@rschmitt I'm surprised IntelliJ would give you a warning about discarding the return value. Is the method you are calling annotated with `@CheckReturnValue `?

Can someone else with a recent version of IntelliJ IDEA verify that IntelliJ gives a warning if you call a method that returns a `<T extends Throwable>` without using the return value? It would be really unfortunate to have to have two methods that are identical except for the method name and whether there is a return value.

As for the method name, I slightly prefer `assertThrows()` vs. `checkThrows()`. The googletest C++ testing framework uses "expect" to mean "if this check values, cause the test to fail, but don't abort to test" (kind of like what `ErrorCollector` does). Also, `assertThrows` would be easier to discover.